apps/io-services-cms-webapp/src/publicator/request-publication-handler.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  FSMStore,
+  ServiceLifecycle,
+  ServicePublication,
+} from "@io-services-cms/models";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+import { Json } from "io-ts-types";
+import { withJsonInput } from "../lib/azure/misc";
+
+const parseIncomingMessage = (
+  queueItem: Json
+): E.Either<Error, ServiceLifecycle.definitions.Service> =>
+  pipe(
+    queueItem,
+    ServiceLifecycle.definitions.Service.decode,
+    E.mapLeft(flow(readableReport, (_) => new Error(_)))
+  );
+
+export class RequestPublicationParseError extends Error {
+  public kind = "RequestPublicationParseError";
+  constructor() {
+    super(`Error while parsing incoming message`);
+  }
+}
+
+export const createRequestPublicationHandler = (
+  store: FSMStore<ServicePublication.ItemType>
+): ReturnType<typeof withJsonInput> =>
+  withJsonInput((_context, queueItem) =>
+    pipe(
+      queueItem,
+      parseIncomingMessage,
+      TE.fromEither,
+      TE.mapLeft((_) => new RequestPublicationParseError()),
+      TE.chain((service) =>
+        ServicePublication.apply("override", service.id, { data: service })(
+          store
+        )
+      ),
+      TE.getOrElse((e) => {
+        throw e;
+      })
```

Comment:
will the error be logged automatically by the az fn? if don't, we have to at lest log it 😉 

PS: @balanza is it possible to add a new field to item before it will be stored to poison queue? I am thinking of storing the error (message, type, or other) to have enough data to analyze the problem directly from the poison queue.

Reply:
I'm pretty sure that _function throw_ is automatically logged by azure.